edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/planner/HealthRestartPlanner.cs

```diff
@@ -41,17 +41,20 @@ public class HealthRestartPlanner : IPlanner
         readonly IEntityStore<string, ModuleState> store;
         readonly TimeSpan intensiveCareTime;
         readonly IRestartPolicyManager restartManager;
+        readonly ICreateUpdateCommandFactory commandMaker;
```

Comment:
I think there shouldn't be 2 factories passed to planner. If you want to keep the ICommandFactory to create just a command, then I think you can embed ICommandFactory into ICreateUpdateCommandFactory and rename it to reflect it creates all command types with prerequired/upfront command.